### PR TITLE
install: implement stripping symbol table

### PR DESCRIPTION
**Pull request by sivachandran**
A simple hello world object file(`helloworld.o`) is added to the fixture for the testing.

Addresses #1759

### CODE REVIEW

File: tests/by-util/test_install.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@ use crate::common::util::*;
 use filetime::FileTime;
 use rust_users::*;
 use std::os::unix::fs::PermissionsExt;
+use std::process::Command;
```
Comment (sylvestre) [inline tests/by-util/test_install.rs:5]:
```suggestion
#[cfg(target_os = "linux")]
use std::process::Command;
```

File: tests/by-util/test_install.rs [hunk 2]
```diff
@@ -566,3 +567,59 @@ fn test_install_copy_then_compare_file_with_extra_mode() {
 
     assert!(after_install_sticky != after_install_sticky_again);
 }
+
+#[test]
+#[cfg(target_os = "linux")]
```
Comment (sylvestre) [inline tests/by-util/test_install.rs:572]: What about mac? I think it should work there too, no?

File: src/uu/install/src/install.rs [hunk 3]
```diff
@@ -164,17 +168,15 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
                 .help("apply access/modification times of SOURCE files to corresponding destination files")
         )
         .arg(
-            // TODO implement flag
             Arg::with_name(OPT_STRIP)
             .short("s")
             .long(OPT_STRIP)
-            .help("(unimplemented) strip symbol tables")
+            .help("strip symbol tables")
```
Comment (sylvestre) [inline src/uu/install/src/install.rs:174]: you should add on which os it is supported
example:
```suggestion
            .help("strip symbol tables (no action Windows)")
```

**Review by sylvestre — CHANGES_REQUESTED** [review]

File: src/uu/install/src/install.rs [hunk 4]
```diff
@@ -171,7 +171,7 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
             Arg::with_name(OPT_STRIP)
             .short("s")
             .long(OPT_STRIP)
-            .help("strip symbol tables")
+            .help("strip symbol tables (no action Windows)")
```
Comment (sylvestre) [inline src/uu/install/src/install.rs:174]: please, same for OPT_STRIP_PROGRAM

File: src/uu/install/src/install.rs [hunk 5]
```diff
@@ -521,6 +525,21 @@ fn copy(from: &PathBuf, to: &PathBuf, b: &Behavior) -> Result<(), ()> {
...
     }
 
+    if b.strip && cfg!(not(windows)) {
+        match Command::new(&b.strip_program).arg(to).output() {
+            Ok(o) => {
+                if !o.status.success() {
+                    crash!(
+                        1,
+                        "strip program failed: {}",
+                        String::from_utf8(o.stderr).unwrap_or_default()
```
Comment (sylvestre) [inline src/uu/install/src/install.rs:535]: can you please trigger this error ?
for example, pass "ls" as strip program
(sorry I didn't notice earlier)

Comment (sivachandran) [reply src/uu/install/src/install.rs:535]: No issues. Added a test case to check this.

File: tests/by-util/test_install.rs [hunk 6]
```diff
@@ -566,3 +568,76 @@ fn test_install_copy_then_compare_file_with_extra_mode() {
...
+
+    let output = Command::new(SYMBOL_DUMP_PROGRAM)
+        .arg("-t")
+        .arg(at.plus(target_file))
+        .output()
+        .unwrap();
+    let stdout = String::from_utf8(output.stdout).unwrap();
+    assert!(!stdout.contains("main"));
+
+    let stderr = scene
```
Comment (sylvestre) [inline tests/by-util/test_install.rs:622]: please move that into a separate test
it helps debugging when tests are focusing on one thing

File: tests/by-util/test_install.rs [hunk 7]
```diff
@@ -566,3 +568,76 @@ fn test_install_copy_then_compare_file_with_extra_mode() {
...
+        .arg("-s")
+        .arg("--strip-program")
+        .arg("/bin/date")
+        .arg(source_file)
+        .arg(target_file)
+        .fails()
+        .stderr;
+    assert!(stderr.contains("strip program failed"));
+
+    let stderr = scene
```
Comment (sylvestre) [inline tests/by-util/test_install.rs:633]: ditto

**Review by sylvestre — CHANGES_REQUESTED** [review]

**Comment by sylvestre** [comment]
Excellent, many thanks!

**Pull request merged**